crates/nu-command/src/commands/dataframe/filter.rs

```diff
@@ -0,0 +1,71 @@
+use crate::prelude::*;
+use nu_engine::WholeStreamCommand;
+use nu_errors::ShellError;
+use nu_protocol::{
+    dataframe::{NuDataFrame, PolarsData},
+    Signature, SyntaxShape, UntaggedValue, Value,
+};
+
+use super::utils::parse_polars_error;
+pub struct DataFrame;
+
+impl WholeStreamCommand for DataFrame {
+    fn name(&self) -> &str {
+        "dataframe filter"
+    }
+
+    fn usage(&self) -> &str {
+        "Filters dataframe using a mask as reference"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("dataframe filter")
+            .required("with", SyntaxShape::String, "the word 'with'")
+            .required("mask", SyntaxShape::Any, "boolean mask used to filter data")
+    }
+
+    fn run(&self, args: CommandArgs) -> Result<OutputStream, ShellError> {
+        command(args)
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: "Filter dataframe using a mask",
+            example: r#"let mask = ([5 6] | dataframe to-series) > 5;
+[[a b]; [1 2] [3 4]] | dataframe to-df | dataframe filter with $mask"#,
```

Comment:
Unfortunately, I'm even more confused now. I'm looking through the pandas documention to understand how this works in pandas and it seems like the mask is the opposite of a where.

```
> s.where(s > 1, 10)
0    10
1    10
2    2
3    3
4    4
dtype: int64

> s.mask(s > 1, 10)
0     0
1     1
2    10
3    10
4    10
dtype: int64
```

I can look at the above examples and understand roughly what's happening, but when I move to this example, I don't understand what's happening.

```
let mask = ([5 6] | dataframe to-series) > 5
```

This is creating a mask that's different than the input values. Is it intended to mean to mask not the first one but the second one?

I don't mean to get stuck on this one example, but I think I've generally been able to understand examples for the dataframes (mostly) so far. I just wasn't able to understand this one.

Reply:
No problem. Let me try to explain what the example is doing. A mask in dataframe terminology is a Series of booleans that can be used to filter a dataframe. This mask can be calculated using any condition that you want to assert and then use that mask to filter a dataframe of the same length. 

The example is doing the next thing:

1.- Creating a mask
```
let mask = ([5 6] | dataframe to-series) > 5
```
Producing a Series of Bools like this one:
```
> $mask
0|false
1|true
```
This indicates that only the second element of the series matches the condition

2. Filter the dataframe with that mask
```
let df = ([[a b]; [1 2] [3 4]] | dataframe to-df | dataframe filter with $mask)
```
Returning
```
> $df 
i|a|b
0|2|4
```
In pandas, the analogous operation would be to use `loc` which can use a mask to return a filtered dataframe

Reply:
@elferherrera got it, that's really helpful.

With that being the case, maybe we should have two examples. The first one could use booleans. Something like:

let mask = ([$true, $false] | dataframe to-series)

And then the second example could use the overloaded comparison that you have now.

Having the two steps makes it easier to understand what the comparison is doing and how it is separate from what the mask is doing.